File: src/test/java/fields/Fields.java
package fields;

import io.github.subiyacryolite.jds.JdsField;
import io.github.subiyacryolite.jds.enums.JdsFieldType;

/**
 * Created by ifunga on 18/02/2017.
 */
public class Fields {
    public static final JdsField STREET_NAME = new JdsField(1, "street_name", JdsFieldType.TEXT);
    public static final JdsField PLOT_NUMBER = new JdsField(2, "plot_number", JdsFieldType.INT);
    public static final JdsField AREA_NAME = new JdsField(3, "area_name", JdsFieldType.TEXT);
    public static final JdsField PROVINCE_NAME = new JdsField(4, "province_name", JdsFieldType.TEXT);
    public static final JdsField CITY_NAME = new JdsField(5, "city_name", JdsFieldType.TEXT);
    public static final JdsField SEX_ENUM = new JdsField(6, "sex_enum", JdsFieldType.ENUM);
    public static final JdsField COUNTRY_NAME = new JdsField(7, "country_name", JdsFieldType.TEXT);
    public static final JdsField PRIMARY_ADDRESS_ENUM = new JdsField(8, "primary_address_enum", JdsFieldType.ENUM);
    public static final JdsField LOCAL_DATE_OF_REGISTRATION = new JdsField(9, "local_date_of_registration", JdsFieldType.DATE_TIME);
    public static final JdsField ZONED_DATE_OF_REGISTRATION = new JdsField(10, "zoned_date_of_registration", JdsFieldType.ZONED_DATE_TIME);
    public static final JdsField STRING_FIELD = new JdsField(1000, "STRING_FIELD", JdsFieldType.TEXT);
    public static final JdsField TIME_FIELD = new JdsField(1009, "TIME_FIELD", JdsFieldType.TIME);
    public static final JdsField DATE_FIELD = new JdsField(1001, "DATE_FIELD", JdsFieldType.DATE);
    public static final JdsField DATE_TIME_FIELD = new JdsField(1002, "DATE_TIME_FIELD", JdsFieldType.DATE_TIME);
    public static final JdsField ZONED_DATE_TIME_FIELD = new JdsField(1003, "ZONED_DATE_TIME_FIELD", JdsFieldType.ZONED_DATE_TIME);
    public static final JdsField LONG_FIELD = new JdsField(1004, "LONG_FIELD", JdsFieldType.LONG);
    public static final JdsField INT_FIELD = new JdsField(1005, "INT_FIELD", JdsFieldType.INT);
    public static final JdsField DOUBLE_FIELD = new JdsField(1006, "DOUBLE_FIELD", JdsFieldType.DOUBLE);
    public static final JdsField FLOAT_FIELD = new JdsField(1007, "FLOAT_FIELD", JdsFieldType.FLOAT);
    public static final JdsField BOOLEAN_FIELD = new JdsField(1008, "BOOLEAN_FIELD", JdsFieldType.BOOLEAN);
    public static final JdsField BLOB_FIELD = new JdsField(1010, "BLOB_FIELD", JdsFieldType.BLOB);
}

File: src/test/java/constants/Fields.kt
package constants

import io.github.subiyacryolite.jds.JdsField
import io.github.subiyacryolite.jds.enums.JdsFieldType

/**
 * Created by ifunga on 18/02/2017.
 */
object Fields {
    val STREET_NAME = JdsField(1, "street_name", JdsFieldType.STRING)
    val PLOT_NUMBER = JdsField(2, "plot_number", JdsFieldType.INT)
    val AREA_NAME = JdsField(3, "area_name", JdsFieldType.STRING)
    val PROVINCE_NAME = JdsField(4, "province_name", JdsFieldType.STRING)
    val CITY_NAME = JdsField(5, "city_name", JdsFieldType.STRING)
    val SEX_ENUM = JdsField(6, "sex_enum", JdsFieldType.ENUM)
    val COUNTRY_NAME = JdsField(7, "country_name", JdsFieldType.STRING)
    val PRIMARY_ADDRESS_ENUM = JdsField(8, "primary_address_enum", JdsFieldType.ENUM)
    val LOCAL_DATE_OF_REGISTRATION = JdsField(9, "local_date_of_registration", JdsFieldType.DATE_TIME)
    val ZONED_DATE_OF_REGISTRATION = JdsField(10, "zoned_date_of_registration", JdsFieldType.ZONED_DATE_TIME)
    val STRING_FIELD = JdsField(11, "string_field", JdsFieldType.STRING)
    val TIME_FIELD = JdsField(12, "time_field", JdsFieldType.TIME)
    val DATE_FIELD = JdsField(13, "date_field", JdsFieldType.DATE)
    val DATE_TIME_FIELD = JdsField(14, "date_time_field", JdsFieldType.DATE_TIME)
    val ZONED_DATE_TIME_FIELD = JdsField(15, "zoned_date_time_field", JdsFieldType.ZONED_DATE_TIME)
    val LONG_FIELD = JdsField(16, "long_field", JdsFieldType.LONG)
    val INT_FIELD = JdsField(17, "int_field", JdsFieldType.INT)
    val DOUBLE_FIELD = JdsField(18, "double_field", JdsFieldType.DOUBLE)
    val FLOAT_FIELD = JdsField(19, "float_field", JdsFieldType.FLOAT)
    val BOOLEAN_FIELD = JdsField(20, "boolean_field", JdsFieldType.BOOLEAN)
    val BLOB_FIELD = JdsField(21, "blob_field", JdsFieldType.BLOB)
    val ADDRESSES = JdsField(22, "addresses", JdsFieldType.ENTITY_COLLECTION)
    val PERIOD = JdsField(23, "period", JdsFieldType.PERIOD)
    val DURATION = JdsField(24, "duration", JdsFieldType.DURATION)
    val MONTH_DAY = JdsField(25, "month_day", JdsFieldType.MONTH_DAY)
    val YEAR_MONTH = JdsField(26, "year_month", JdsFieldType.YEAR_MONTH)
    val RIGHTS= JdsField(27, "rights", JdsFieldType.ENUM_COLLECTION)
}
